farm-staking/farm-staking-proxy/src/lib.rs

```diff
@@ -230,47 +203,29 @@ pub trait FarmStakingProxy:
             pair_second_token_min_amount,
         );
 
-        let staking_farm_token_amount =
+        let full_staking_farm_token_amount =
             self.get_staking_farm_token_amount_equivalent(&payment_amount);
+        let staking_farm_exit_amount = self.get_staking_farm_token_amount_equivalent(&exit_amount);
         let staking_farm_exit_result = self.staking_farm_unstake(
             remove_liq_result.staking_token_payment,
             attributes.staking_farm_token_nonce,
-            staking_farm_token_amount,
-        );
-        let unstake_result = self.send_unstake_payments(
-            remove_liq_result.other_token_payment,
-            lp_farm_exit_result.lp_farm_rewards,
-            staking_farm_exit_result.staking_rewards,
-            staking_farm_exit_result.unbond_staking_farm_token,
+            full_staking_farm_token_amount,
+            staking_farm_exit_amount,
         );
 
-        self.burn_dual_yield_tokens(payment_nonce, &payment_amount);
-
-        unstake_result
-    }
-
-    fn send_unstake_payments(
-        &self,
-        other_token_payment: EsdtTokenPayment<Self::Api>,
-        lp_farm_rewards: EsdtTokenPayment<Self::Api>,
-        staking_rewards: EsdtTokenPayment<Self::Api>,
-        unbond_staking_farm_token: EsdtTokenPayment<Self::Api>,
-    ) -> UnstakeResult<Self::Api> {
         let caller = self.blockchain().get_caller();
-        let mut user_payments = ManagedVec::new();
-        if other_token_payment.amount > 0 {
-            user_payments.push(other_token_payment);
-        }
-        if lp_farm_rewards.amount > 0 {
-            user_payments.push(lp_farm_rewards);
-        }
-        if staking_rewards.amount > 0 {
-            user_payments.push(staking_rewards);
-        }
-        user_payments.push(unbond_staking_farm_token);
-
-        self.send().direct_multi(&caller, &user_payments);
-
-        user_payments.into()
+        let remaining_dual_yield_tokens =
+            EsdtTokenPayment::new(payment_token, payment_nonce, &payment_amount - &exit_amount);
```

Comment:
There seems to be a wrong usage of parameters. Looks like exit_amount refers to the farm position, and it should not be deducted from the payment amount.

Reply:
You're right, the current logic is wrong. Will merge with the fix once it enters the main branch.

Reply:
Fixed.